sdk/aqueduct/api_client.py

```diff
@@ -132,13 +135,22 @@ def __init__(self, api_key: str, aqueduct_address: str):
         else:
             self.use_https = self._test_connection_protocol(try_http=True, try_https=True)
 
+    def _check_config(self) -> None:
+        if not self.configured:
+            raise Exception(
+                "API client has not been configured, please complete the configuration \
+                by initializing an Aqueduct client with the api key and the server address."
+            )
```

Comment:
Instead of having to label every single helper in this class, I think all we need to check configs for is:
`construct_base_url()`, `construct_full_url()`, `url_prefix()`, and `generate_auth_headers()`. The reason I think this makes more sense is because each of these helpers directly uses the methods we've set in `configure()`.

It would also be great to move `generate_auth_headers()` into the APIClient, I'm not sure why these were in utils to begin with.